parser_example_test.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parser_test
+
+import (
+	"fmt"
+	"github.com/pingcap/parser"
+	// 0. import parser_driver implemented by TiDB(user also can implement own driver by self).
+	_ "github.com/pingcap/tidb/types/parser_driver"
+)
+
+// This example show how to parse a text sql into ast.
+func Example_parseSQL() {
+
+	// 0. make sure import parser_driver implemented by TiDB(user also can implement own driver by self).
+	// import _ "github.com/pingcap/tidb/types/parser_driver" in import part.
+
+	// 1. Create a parser, this is a NOT thread-safe but heavy object,
+	// it is better to reuse it in thread-safe way as possible  as we can.
```

Comment:
I don't understand, do you mean e.g. use a `sync.Mutex` to protect it if you want to access the parser from multiple threads?

Reply:
Each goroutine should use it's own parser @kennytm 

Don't do this:
```
p := parser.New()
go func() { p.Parse(...) } ()
go func() { p.Parse(...) } ()
```

Reply:
Thanks @tiancaiamao. The comment should be rephrased to not use the word "reuse" then. Maybe you mean "move"?

Reply:
@kennytm  I mean "reuse it in thread-local level", but how to say it :rofl:

Reply:
@lysu  I would split it into two sentences.

```go
// 1. Create a parser. The parser is NOT goroutine safe and should
// not be shared among multiple goroutines. However, parser is also
// heavy, so each goroutine should reuse its own local instance if 
// possible.
```